### ISSUE TITLE
Regression: Mapbox Voice Instructions

### ISSUE THREAD

**Issue by JThramer** [mapbox/mapbox-navigation-ios#1634]
Hey all,

It looks like we introduced a regression onto `master` as part of #1596. It looks like an `NSException` is being thrown when we try to [prepare the audio player for playing.](https://github.com/mapbox/mapbox-navigation-ios/blob/master/MapboxNavigation/MapboxVoiceController.swift?utf8=✓#L225)

### Console Log
```
2018-08-29 14:22:15.439976-0600 Example-Swift[56466:8459466] [AudioHAL_Client] HALC_ProxyIOContext.cpp:1399:IOWorkLoop:  HALC_ProxyIOContext::IOWorkLoop: failed to send the final message to the server, Error: 0x10000003
2018-08-29 14:22:15.451383-0600 Example-Swift[56466:8459641] [AudioHAL_Client] HALC_ProxyIOContext.cpp:959:IOWorkLoop:  HALC_ProxyIOContext::IOWorkLoop: failed to get the IOPort
2018-08-29 14:22:30.504840-0600 Example-Swift[56466:8459164] 239: CA_UISoundClientBase::StartPlaying: AddRunningClient failed (status = -66681).
```
### Expected Behavior:
* Polly to speak voice instructions.

### Observed Behavior:
* In `Example-Swift`, Polly speaks the first instruction, `AVSpeechSynthesizer` speaks the remaining voice instructions for the route.
* In `Example-Objective-C`, both voice controllers speak simultaneously.

/cc @mapbox/navigation-ios

**Comment by bsudekum** [comment]
I'm unable to reproduce this on master. @JThramer let me know if you have any clues as to how to trigger this error.

**Comment by JThramer** [comment]
Yeah there is definitely a weird data-race going on, @bsudekum. I *just* got a run where both voices spoke simultaneously  (in `Example-Swift`, no less), but right as I go to debug, the LLDB RPC server crashes.

MFW: 
![seriously?](https://user-images.githubusercontent.com/887225/44817164-cb147500-aba1-11e8-985c-2f8abc683d78.gif)


Still investigating.

**Comment by JThramer** [comment]
@bsudekum I definitely found the source of the "[doubling up](https://github.com/mapbox/mapbox-navigation-ios/blob/master/Examples/Objective-C/ViewController.m#L69)" of voices on the OBJ-C sample. The rest I'm thinking could be cache pollution, but it could also be mental fatigue. 😅
